src/models/DraftViews.ts

```diff
@@ -79,8 +79,16 @@ class DraftViews {
                 this.hostEvents.push(specMessage);
             }
             if (draftEvent.executingPlayer === Player.NONE) {
-                this.guestEvents.push(draftEvent);
-                this.hostEvents.push(draftEvent);
+                if (draftEvent.player === Player.HOST) {
+                    this.hostEvents.push(draftEvent);
+                    this.guestEvents.push(specMessage);
+                } else if (draftEvent.player === Player.GUEST) {
+                    this.guestEvents.push(draftEvent);
+                    this.hostEvents.push(specMessage);
+                } else {
+                    this.guestEvents.push(draftEvent);
+                    this.hostEvents.push(draftEvent);
+                }
```

Comment:
Wouldn't it make sense to just push the specMessage to both here?

Reply:
That's what I initially wanted, but that's cause issues.
- non-player actions can't be set to HIDDEN, Validations don't allow it, and for good reason.
- If admin actions were hidden, it wouldn't be able to see it's own actions, because it uses guestEvents (https://github.com/SiegeEngineers/aoe2cm2/blob/master/src/models/Validation.ts#L84) in those steps. And thus it could BAN/PICK the same civ multiple times at random.

To make it work you'd need a separate `this.adminEvents` and update the validations to use that for admin picks/bans, but imo there's no reason to hide admin actions except potentially 'dramatic effect'.

Reply:
Hm I guess that makes sense